modules/agentless_scanner_all_resources/main.tf

```diff
@@ -0,0 +1,368 @@
+
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+locals {
+  account_id  = data.aws_caller_identity.current.account_id
+  region_code = data.aws_region.current.name
+  tags        = merge(var.additional_tags, { "CreatedBy" = "${var.resources_prefix}-agentless" })
+  ami_map = {
+    "us-west-2" : "ami-02d5622660e852be0"
+    "us-west-1" : "ami-0eb917bd5c3b95e02"
+    "us-east-2" : "ami-0fddca2efe6e54d98"
+    "us-east-1" : "ami-07719efefe896200a"
+  }
+}
+
+#iam resources -------------------------------------------------
+
+locals {
+  scanner_iam_role_name = "${var.resources_prefix}-agentless-scanner"
+  monitor_iam_role_name = "${var.resources_prefix}-agentless-monitor"
+  target_iam_role_name  = "${var.resources_prefix}-agentless-target"
+
+  scanner_instance_profile_name = "${var.resources_prefix}-agentless-scanner-profile"
+  monitor_instance_profile_name = "${var.resources_prefix}-agentless-monitor-profile"
+
+  scanner_iam_role_arn = "arn:aws:iam::${local.account_id}:role/${local.scanner_iam_role_name}"
+}
+
+resource "aws_iam_role" "uptycs_scanner_role" {
+  count = var.is_default_region ? 1 : 0
+
+  name                 = local.scanner_iam_role_name
+  assume_role_policy   = templatefile("${path.module}/../../templates/agentless-scanner-trust-policy.json.tpl", {})
+  tags                 = local.tags
+  permissions_boundary = var.scanner_role_pb_policy_arn
+
+  inline_policy {
+    name = "UptycsAgentlessScannerPolicy"
```

Comment:
Name can't be static, otherwise QA will have an issue.

Reply:
This is inline policy. There will be no conflict.